### PR TITLE
session: move err_rollback test.

### PR DESCRIPTION
**Pull request by hanfei1991**
We should change the storage of test to mocktikv step by step. But currently the test cannot pass, maybe some bugs exist in mocktikv.

### CODE REVIEW

File: new_session_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	tk.MustExec("create table t_rollback (c1 int, c2 int, primary key(c1))")
+	tk.MustExec("insert into t_rollback values (0, 0)")
+
+	var wg sync.WaitGroup
+	cnt := 4
+	wg.Add(cnt)
+	num := 100
+
+	// retry forever
+	tidb.SetCommitRetryLimit(math.MaxInt64)
```
Comment (coocood) [inline new_session_test.go]: Why not just sets the `unlimitedRetryCount ` as the old test code?

Comment (hanfei1991) [reply new_session_test.go]: This test is in `tidb_test` package, for using mock-tikv and testkit.

File: new_session_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/store/tikv"
+	"github.com/pingcap/tidb/store/tikv/mock-tikv"
+	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/util/testleak"
+)
+
+var _ = Suite(&testSessionSuite{})
```
Comment (tiancaiamao) [inline new_session_test.go]: `testSessionSuite` already exists?

File: new_session_test.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	tidb.SetCommitRetryLimit(math.MaxInt64)
+	defer tidb.SetCommitRetryLimit(10)
+
+	for i := 0; i < cnt; i++ {
+		go func() {
+			defer wg.Done()
+			localTk := testkit.NewTestKit(c, s.store)
+			localTk.MustExec("use test")
+			for j := 0; j < num; j++ {
+				localTk.Exec("insert into t_rollback values (1, 1)")
```
Comment (tiancaiamao) [inline new_session_test.go]: what's the purpose of this line?

File: session.go [hunk 4]
```diff
@@ -755,41 +755,6 @@ func (s *session) DropPreparedStmt(stmtID uint32) error {
 	return nil
 }
 
-// GetTxn gets the current transaction or creates a new transaction.
-// If forceNew is true, GetTxn() must return a new transaction.
-// In this situation, if current transaction is still in progress,
-// there will be an implicit commit and create a new transaction.
-func (s *session) GetTxn(forceNew bool) (kv.Transaction, error) {
```
Comment (tiancaiamao) [inline session.go]: Why remove this?

**Comment by tiancaiamao** [comment]
Is the session auto-committed?

**Comment by coocood** [comment]
@tiancaiamao 
Yes, it is.

File: new_session_test.go (hunk 2 above)
Comment (hanfei1991) [reply new_session_test.go]: this is a new test package.

**Comment by coocood** [comment]
The result is expected to be 400, but the actual is a little less than 400, every time has a different result.

File: new_session_test.go (hunk 3 above)
Comment (hanfei1991) [reply new_session_test.go]: move from old code, this sql may fail and roll back.

File: session.go (hunk 4 above)
Comment (hanfei1991) [reply session.go]: This function is not needed any longer

**Comment by tiancaiamao** [comment]
Bug fixed, it happens just in mocktikv,  TiKV is not involved.
We check data before err here: https://github.com/pingcap/tidb/pull/4412/files#diff-9f657f87f877075313b8774ebe81ae86L85
`update xxx where xxx`  should meet error locked and retry during reading data, but it doesn't read anything, so it update nothing and the update operation is lost.

@zimulala @coocood

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**